src/MCPServer/lib/RPCServer.py

```diff
@@ -54,206 +65,374 @@ class NotFoundError(RPCServerError):
+         Disable ``raise_exc`` when you want the client to receive a detailed
+         error.
+
+    3. Handler methods have at least three arguments:
+
+       - ``self``: instance of the RPCServer class.
+
+       - ``worker``: instance of the worker class which is a subclass of
+         ``gearman.job.GearmanWorker``.
+
+       - ``job``: instance of ``gearman.job.GearmanJob``.
+
+       Additionally, a method handler can have an argument named ``payload``.
+       If present, it will carry the data embedded in the payload of the job
+       already unserialized. This is useful when the handler wants to read
+       data sent by the client.
 
-    If ``raise_exc`` is True, handler exceptions are raised. This is
-    interpreted by the worker library as a failed job and the client will not
-    receive additional information about the error other than its status.
-    Set ``raise_exc`` to False when you want the client to receive a detailed
-    error.
     """
-    def decorator(func):
-        def wrap(*args, **kwargs):
+    # Regular expression to match handler configurations in method docstrings.
+    ability_regex = re.compile(
+        r"(?P<docstring>.*)(?P<config>\[config\].*)$",
+        re.DOTALL | re.MULTILINE)
+
+    # To some extent, application logic in Archivematica is coupled to certain
+    # values in our workflow data. This is mostly true to things related to
+    # types of transfers and its approvals. This value in particular supports
+    # the partial approval of AIP reingest.
+    APPROVE_AIP_REINGEST_CHAIN_ID = "9520386f-bb6d-4fb9-a6b6-5845ef39375f"
+
+    def __init__(self, workflow):
+        super(RPCServer, self).__init__(
+            host_list=[django_settings.GEARMAN_SERVER])
+        self.workflow = workflow
+        self._register_tasks()
+        self.set_client_id(gethostname() + "_MCPServer")
+
+    def _register_tasks(self):
+        for ability, handler in self._handlers():
+            logger.debug("Registering ability %s", ability)
+            self.register_task(ability, handler)
+
+    def _handlers(self):
+        members = inspect.getmembers(self, predicate=inspect.ismethod)
+        for name, handler in members:
+            if name[0] != "_":
+                continue
+            if not name.endswith("_handler"):
+                continue
+            name, opts = self._get_ability_details(handler)
+            yield name, self._wrap_handler_method(handler, **opts)
+
+    def _get_ability_details(self, handler):
+        docstring = str(inspect.getdoc(handler))
+        match = self.ability_regex.search(docstring)
+        err = ValueError("Unexpected docstring format: {}".format(docstring))
+        if match is None:
+            raise err
+        try:
+            config = match.group("config")
+        except IndexError:
+            raise err
+        parser = configparser.SafeConfigParser({
+            "name": None, "raise_exc": False,
```

Comment:
It's a small thing, but `"raise_exc": False` are all set to `no` or `yes` below. Maybe the Python `True/False` style would be a little more natural?

Reply:
Well if you ask me, I think that there's nothing more natural than natural languages.

Reply:
I changed this to `True`/`False` as requested, forgot to mention it.